src/libraries/Reserves.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library Reserves {
+    /// uint256(keccak256("ReservesOf")) - 1
+    uint256 constant RESERVES_OF_SLOT = uint256(0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95);
+    /// @notice The transient reserves for pools with no balance is set to the max as a sentinel to track that it has been synced.
+    uint256 public constant ZERO_BALANCE = type(uint256).max;
+
+    /// @notice Thrown when someone has not called sync before calling settle for the first time.
+    error ReservesMustBeSynced();
+
+    function setReserves(Currency currency, uint256 value) internal {
+        if (value == 0) value = ZERO_BALANCE;
```

Comment:
i think we need a  `if (value == ZERO_BALANCE) revert()`
because...
- contract has `balance==type(uint256).max`
- call `sync` which passes in `type(uint256).max` which gets stored
- call `settle` immediately which performs `reservesNow - reservesBefore` which is `type(uint256).max - 0` and then settles `type(uint256).max` to the caller

i dont really see anything you can do with that attack because you'd basically have to mint `max` to the contract and then youd be able to claim them... feels pointless. But I think we should add the check.

Reply:
Yeah interesting - this makes me want to just use a different implementation of storing a flag if somethings been synced or not

Reply:
Ok I believe we are deeming this a non issue because we assume that token balances are less than `type(uint128).max `